src/main/java/com/digitaldan/jomnilinkII/examples/Main.java

```diff
@@ -28,240 +28,319 @@
+                    ObjectProperties.FILTER_2_NONE, ObjectProperties.FILTER_3_NONE))
+                            .getMessageType() == Message.MESG_TYPE_OBJ_PROP) {
+                logger.info(m.toString());
+                objnum = ((ObjectProperties) m).getNumber();
+            }
+            objnum = 0;
+            while ((m = c.reqObjectProperties(Message.OBJ_TYPE_CONTROL_READER, objnum, 1,
+                    ObjectProperties.FILTER_1_NAMED, ObjectProperties.FILTER_2_NONE, ObjectProperties.FILTER_3_NONE))
+                            .getMessageType() == Message.MESG_TYPE_OBJ_PROP) {
+                logger.info(m.toString());
+                objnum = ((ObjectProperties) m).getNumber();
+            }
 
-			logger.info(c.uploadNames(Message.OBJ_TYPE_UNIT, 0).toString());
-			logger.info(c.reqSecurityCodeValidation(1, 1, 2, 3, 4).toString());
+            ObjectStatus status = c.reqObjectStatus(Message.OBJ_TYPE_ZONE, 1, max_zones);
+            ZoneStatus[] zones = (ZoneStatus[]) status.getStatuses();
+            for (int i = 0; i < zones.length; i++) {
+                logger.info(zones[i].toString());
+            }
+            status = c.reqObjectStatus(Message.OBJ_TYPE_ZONE, 1, max_zones, true);
+            ExtendedZoneStatus[] zonesExt = (ExtendedZoneStatus[]) status.getStatuses();
+            for (int i = 0; i < zonesExt.length; i++) {
+                logger.info(zonesExt[i].toString());
+            }
+            status = c.reqObjectStatus(Message.OBJ_TYPE_UNIT, 1, max_units);
+            UnitStatus[] units = (UnitStatus[]) status.getStatuses();
+            for (int i = 0; i < units.length; i++) {
+                logger.info(units[i].toString());
+            }
+            status = c.reqObjectStatus(Message.OBJ_TYPE_UNIT, 1, max_units, true);
+            ExtendedUnitStatus[] unitsExt = (ExtendedUnitStatus[]) status.getStatuses();
+            for (int i = 0; i < units.length; i++) {
+                logger.info(unitsExt[i].toString());
+            }
+            status = c.reqObjectStatus(Message.OBJ_TYPE_AREA, 1, max_areas);
+            AreaStatus[] areas = (AreaStatus[]) status.getStatuses();
+            for (int i = 0; i < areas.length; i++) {
+                logger.info(areas[i].toString());
+            }
+            status = c.reqObjectStatus(Message.OBJ_TYPE_AREA, 1, max_areas, true);
+            ExtendedAreaStatus[] areasExt = (ExtendedAreaStatus[]) status.getStatuses();
+            for (int i = 0; i < areasExt.length; i++) {
+                logger.info(areasExt[i].toString());
+            }
+            status = c.reqObjectStatus(Message.OBJ_TYPE_THERMO, 1, max_thermos);
+            ThermostatStatus[] thermos = (ThermostatStatus[]) status.getStatuses();
+            for (int i = 0; i < thermos.length; i++) {
+                logger.info(thermos[i].toString());
+            }
+            status = c.reqObjectStatus(Message.OBJ_TYPE_THERMO, 1, 16, true); // It appears that the maximum number of
```

Comment:
is this a testing line ?  its throwing an exception on my system since i don't have thermostats on my omni